samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -105,15 +105,19 @@
    * <p>
    * Use the {@link org.apache.samza.operators.windows.Windows} helper methods to create the appropriate windows.
    * <p>
-   * <b>Warning:</b> As of version 0.13.0, messages in windows are kept in memory and will be lost during restarts.
+   * The {@code id} is used as part of the unique ID for any state stores and streams created by this transform.
```

Comment:
In the test code, it seems that the same id + different opCode is allowed in the same application. It would be nice to clarify what "part of the unique ID" means here. Are we explicitly asking user to supply a fully-qualified unique ID in the application here as id? Or just unique ID for each type of transform in his application? Either way, need to be explicit here. The implementation may not need to change at this point, but the user expectation needs to be set clearly.

Reply:
I'd prefer just saying "unique in the application", since that's easy to understand and implies unique ID per operator type. Will clarify in documentation.